tests/models/dac/test_modeling_dac.py

```diff
@@ -0,0 +1,735 @@
+
+        inputs = processor(
+            raw_audio=audio_samples,
+            sampling_rate=processor.sampling_rate,
+            truncation=False,
+            return_tensors="pt",
+        ).to(torch_device)
+
+        with torch.no_grad():
+            encoder_outputs = model.encode(inputs["input_values"])
+
+            expected_encoder_sums = torch.tensor(list(expected_encoder_sums_dict.values()), dtype=torch.float32)
+            encoder_outputs_sum = torch.tensor([v.float().mean().cpu().item() for v in encoder_outputs.to_tuple()])
+
+            # make sure audio encoded codes are correct
+            self.assertTrue(torch.allclose(encoder_outputs_sum, expected_encoder_sums, atol=1e-3))
+
+            _, quantized_representation, _, _ = encoder_outputs.to_tuple()
+            input_values_dec = model.decode(quantized_representation)[0]
+            input_values_enc_dec = model(inputs["input_values"])[1]
+
+            # make sure forward and decode gives same result
+            self.assertTrue(torch.allclose(input_values_dec, input_values_enc_dec, atol=1e-3))
+
+            arr = inputs["input_values"].cpu().numpy()
+            arr_enc_dec = input_values_enc_dec.cpu().numpy()
+
+            max_length = min(arr_enc_dec.shape[-1], arr.shape[-1])
+
+            arr_cut = arr[:, 0, :max_length].copy()
+            arr_enc_dec_cut = arr_enc_dec[:, :max_length].copy()
+
+            # make sure audios are more or less equal
+            rmse = compute_rmse(arr_cut, arr_enc_dec_cut)
+            self.assertTrue(rmse < expected_rmse)
+
+    def test_integration_batch_44khz(self):
+        expected_rmse = 0.001
+
+        expected_encoder_sums_dict = {
+            "loss": 25.9233,
+            "quantized_representation": 0.0013,
+            "audio_codes": 528.5620,
+            "projected_latents": -0.1194,
+        }
+
+        librispeech_dummy = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
+
+        model_name = "dac_44khz"
+
+        model_id = "descript/{}".format(model_name)
+        model = DacModel.from_pretrained(model_id).to(torch_device)
+        processor = AutoProcessor.from_pretrained(model_id)
+
+        librispeech_dummy = librispeech_dummy.cast_column("audio", Audio(sampling_rate=processor.sampling_rate))
+
+        audio_samples = [np.array([audio_sample["array"]])[0] for audio_sample in librispeech_dummy[-2:]["audio"]]
+
+        inputs = processor(
+            raw_audio=audio_samples,
+            sampling_rate=processor.sampling_rate,
+            truncation=False,
+            return_tensors="pt",
+        ).to(torch_device)
+
+        with torch.no_grad():
+            encoder_outputs = model.encode(inputs["input_values"])
+
+            expected_encoder_sums = torch.tensor(list(expected_encoder_sums_dict.values()), dtype=torch.float32)
+            encoder_outputs_sum = torch.tensor([v.float().mean().cpu().item() for v in encoder_outputs.to_tuple()])
```

Comment:
Two things here: 
* The object `expected_encoder_sums_dict` should be renamed to reflect what's being tested
* If you're finding that you're unable to replicate numbers in runs, I would recommend trying just to see if you can replicate just the first set of values in the tuple for a given tolerance.